TypeSystem/Reflection/test.cc

```diff
@@ -118,6 +117,101 @@ static_assert(std::is_same<current::reflection::FieldTypeWrapper<Foo>,
 
 }  // namespace reflection_test
 
+TEST(Reflection, CurrentTypeName) {
+  using current::reflection::CurrentTypeName;
+  using current::reflection::NameFormat;
+  using namespace reflection_test;
+
+  // NameFormat::Z: C++ types, template names as `_Z`, variant names expanded as `_B_..._E`.
+  // Unused for the very naming, kept for legacy reasons of inner TypeID computation when the dependency is cyclic (?).
+  // TODO(dkorolev) + TODO(mzhurovich): Why don't we retire this? It would change some TypeIDs, so what?
```

Comment:
At least we have to think about remote stream schema ID check that will start failing.